unittests/AST/ASTImporterFixtures.cpp

```diff
@@ -115,19 +115,20 @@ ASTImporterTestBase::TU::TU(StringRef Code, StringRef FileName, ArgVector Args)
 }
 
 void ASTImporterTestBase::TU::lazyInitImporter(
-    ASTImporterSharedState &SharedState, ASTUnit *ToAST) {
+    std::shared_ptr<ASTImporterSharedState> &SharedState, ASTUnit *ToAST) {
   assert(ToAST);
   if (!Importer) {
     Importer.reset(new ASTImporter(
         ToAST->getASTContext(), ToAST->getFileManager(), Unit->getASTContext(),
-        Unit->getFileManager(), false, &SharedState));
+        Unit->getFileManager(), false, SharedState));
   }
   assert(&ToAST->getASTContext() == &Importer->getToContext());
   createVirtualFileIfNeeded(ToAST, FileName, Code);
 }
 
-Decl *ASTImporterTestBase::TU::import(ASTImporterSharedState &SharedState,
-                                      ASTUnit *ToAST, Decl *FromDecl) {
+Decl *ASTImporterTestBase::TU::import(
+    std::shared_ptr<ASTImporterSharedState> &SharedState, ASTUnit *ToAST,
```

Comment:
& is not needed here

Reply:
Yes, we could copy the shared_ptr, but that would mean an unnecessary increment and decrement of the shared count. This is why I choose to pass by ref.

Reply:
But it could be a `const &`? (The pointer itself is not modified?)

Reply:
Yes, that's right, I'll make it a `const &`.